src/lib.rs

```diff
@@ -64,18 +68,63 @@ impl Optimizer {
         args
     }
 
+    /// Produces a tuple value from the string arguments
+    fn parse_arguments(args: Vec<String>) -> Value {
+        Value::Tuple(
+            args.into_iter()
+                .map(|arg| {
+                    if let Ok(int) = arg.parse::<i64>() {
+                        Value::Const(Constant::Int(int))
+                    } else if arg == "true" {
+                        Value::Const(Constant::Bool(true))
+                    } else if arg == "false" {
+                        Value::Const(Constant::Bool(false))
+                    } else {
+                        panic!("Invalid argument to bril program: {}", arg);
+                    }
+                })
+                .collect(),
+        )
+    }
+
+    /// Interpret a program in an `Interpretable` IR.
+    /// Returns the printed output of the program.
+    /// The program should not return a value.
+    pub fn interp(
+        program: &Interpretable,
+        args: Vec<String>,
+        profile_out: Option<PathBuf>,
+    ) -> String {
+        match program {
+            Interpretable::Bril(program) => Self::interp_bril(program, args, profile_out),
+            Interpretable::TreeProgram(program) => {
+                let (val, mut printed) = interpret_tree_prog(program, Self::parse_arguments(args));
+                assert_eq!(val, Value::Tuple(vec![]));
+                // add new line to the end of each line in printed
+                for line in printed.iter_mut() {
+                    line.push('\n');
+                }
+                printed.join("")
```

Comment:
is this different than let mut res = printed.join("\n"); printed.push('\n'); ?

Reply:
Yep- in the case that there is nothing printed